appsec/appsec.go

```diff
@@ -34,6 +39,61 @@ func MonitorParsedHTTPBody(ctx context.Context, body interface{}) {
 	// bonus: use sync.Once to log a debug message once if AppSec is disabled
 }
 
+type userMonitoringError struct {
+	err    error
+	status struct {
+		grpc codes.Code
+		http int
+	}
+}
+
+func (err *userMonitoringError) ShouldBlock() bool {
+	return err.status.grpc == codes.Aborted || err.status.http == 403
+}
+func (err *userMonitoringError) GRPCStatus() *status.Status {
+	return status.New(err.status.grpc, err.Error())
+}
+func (err *userMonitoringError) GRPCCode() codes.Code {
+	return err.status.grpc
+}
+func (err *userMonitoringError) HTTPCode() int {
+	return err.status.http
+}
+func (err *userMonitoringError) Error() string {
+	return err.err.Error()
+}
+
+// SetUser associates user information to the current trace which the
```

Comment:
Explain that we both monitor AND block users and that this SDK should be used when the app is dealing with authenticated users.